lib/http/uri.rb

```diff
@@ -110,6 +109,54 @@ def hash
       @hash ||= to_s.hash * -1
     end
 
+    # Host, either a domain name or IP address. If the host is an IPv6 address, it will be returned
+    # without brackets surrounding it.
+    #
+    # @return [String] The host of the URI
+    def host
+      @host ||= begin
+        ip = IPAddr.new(@uri.host)
+
+        if ip.ipv6?
+          ip.to_s
+        else
+          @uri.host
+        end
+      rescue IPAddr::Error
+        @uri.host
+      end
+    end
+
+    # Normalized host, either a domain name or IP address. If the host is an IPv6 address, it will
+    # be returned without brackets surrounding it.
+    #
+    # @return [String] The normalized host of the URI
+    def normalized_host
+      @normalized_host ||= begin
+        ip = IPAddr.new(@uri.normalized_host)
+
+        if ip.ipv6?
+          ip.to_s
+        else
+          @uri.normalized_host
+        end
+      rescue IPAddr::Error
+        @uri.normalized_host
+      end
+    end
+
+    # Sets the host component for the URI.
+    #
+    # @param [String, #to_str] new_host The new host component.
+    # @return [void]
+    def host=(new_host)
+      @uri.host = new_host
+
+      # Reset dependent values
+      remove_instance_variable(:@host) if defined?(@host)
+      remove_instance_variable(:@normalized_host) if defined?(@normalized_host)
+    end
```

Comment:
It seems like all of this could be done eagerly in the constructor rather than lazily which would simplify the implementation.

Also I'm not sure I see the value of having two methods. I'd suggest just always having `host` return the unbracketed version.

Reply:
Aah, sorry, my bad. Offhand I don’t know what `normalized_host` does either